lte/gateway/c/core/oai/tasks/amf/amf_app_ue_context.cpp

```diff
@@ -407,4 +407,55 @@ int amf_idle_mode_procedure(amf_context_t* amf_ctx) {
   OAILOG_FUNC_RETURN(LOG_AMF_APP, RETURNok);
 }
 
+/****************************************************************************
+ **                                                                        **
+ ** Name:    amf_delete_ue_context()                                       **
+ **                                                                        **
+ ** Description: Deletes the ue context                                    **
+ **                                                                        **
+ **                                                                        **
+ ***************************************************************************/
+void amf_delete_ue_context(ue_m5gmm_context_s* ue_context_p) {
+  hashtable_rc_t h_rc                = HASH_TABLE_OK;
+  amf_app_desc_t* amf_app_desc_p     = get_amf_nas_state(false);
+  amf_ue_context_t* amf_ue_context_p = &amf_app_desc_p->amf_ue_contexts;
+
+  hash_table_ts_t* amf_state_ue_id_ht = get_amf_ue_state();
+  if (!ue_context_p || !amf_ue_context_p) {
+    return;
+  }
+
+  amf_remove_ue_context(ue_context_p);
```

Comment:
If this function is used to delete the UE entry on redis, I would name this `amf_remove_ue_state_from_db` or `amf_remove_ue_context_from_db`

Reply:
this deletes UE entries from hashlists, maps, and de-allocation of memory from heap etc